src/mrack/providers/provider.py

```diff
@@ -140,6 +140,10 @@ async def provision_hosts(self, reqs):
         for host in success_hosts:
             logger.info(f"{self.dsp_name}: {host}")
 
+        if error_hosts:
+            # Forcing exit code to be an error
+            raise ProvisioningError("Some hosts couldn't be provisioned")
```

Comment:
This will leave successfully provisioned hosts provisioned. We should roll-back and destroy all hosts in this case.

Reply:
The deletion of hosts should "in theory" happen already in the
```python
       if error_hosts:
            hosts_to_delete = success_hosts + error_hosts
            await self.abort_and_delete(hosts_to_delete, error_hosts)
```
In the method there is:
```python
    async def abort_and_delete(self, hosts_to_delete, error_hosts):
        """Delete hosts and abort provisioning with an error."""
        logger.info(f"{self.dsp_name}: Aborting provisioning due to error.")
        for host in error_hosts:
            logger.error(f"{self.dsp_name}: Error: {str(host.error)}")

        logger.info(f"{self.dsp_name}: Given the error, will delete hosts")
        await self.delete_hosts(hosts_to_delete)
        raise ProvisioningError(error_hosts)
```

So something else might not be working.

Reply:
Could you please rebase @netoarmando ?

Reply:
I've re-based it and removed this commit after a discussion with @Tiboris.
We agreed that changes made since last release may have fixed this issue.

I also included a simple sleep between retries.